docs/pbc/all/dynamic-multistore/202404.0/tutorials-and-howtos/how-to-add-new-custom-field.md

```diff
@@ -0,0 +1,329 @@
+
+For adjuste the form in the Back Office need to extend `StoreContextFormExpander` and add new field to the form.
+
+First, need to extend `StoreContextForm` and add new field to the form.
+
+**src/Pyz/Zed/StoreContextGui/Communication/Form/StoreContextForm.php**
+
+```php
+
+namespace Pyz\Zed\StoreContextGui\Communication\Form;
+
+use Spryker\Zed\StoreContextGui\Communication\Form\StoreContextForm as SprykerStoreContextForm;
+use Symfony\Component\Form\Extension\Core\Type\EmailType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints\Email;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+class StoreContextForm extends SprykerStoreContextForm
+{
+    /**
+     * @var string
+     */
+    protected const FIELD_SUPPORT_CONTACT_EMAIL = 'contactEmail';
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array<string, mixed> $options
+     *
+     * @return void
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        parent::buildForm($builder, $options);
+
+        $this->addContactEmailField($builder, $options);
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array<string, mixed> $options
+     *
+     * @return \Spryker\Zed\StoreContextGui\Communication\Form\StoreContextForm
+     */
+    protected function addContactEmailField(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add(static::FIELD_SUPPORT_CONTACT_EMAIL, EmailType::class, [
+                'label' => 'Email',
+                'constraints' => $this->createEmailConstraints(),
+            ]);
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    protected function createEmailConstraints(): array
+    {
+        return  [
+            new Email(),
+        ];
+    }
+}
+
+```
+
+Next step is to extend `StoreContextCollectionForm` and use new `StoreContextForm` in the `StoreContextCollectionForm`.
+
+*src/Pyz/Zed/StoreContextGui/Communication/Form/StoreContextCollectionForm.php*
+
+```php
+<?php
+
+namespace Pyz\Zed\StoreContextGui\Communication\Form;
+
+use Pyz\Zed\StoreContextGui\Communication\Form\StoreContextForm;
+use Spryker\Zed\StoreContextGui\Communication\Form\StoreContextCollectionForm as SprykerStoreContextCollectionForm;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\FormBuilderInterface;
+
+class StoreContextCollectionForm extends SprykerStoreContextCollectionForm
+{
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array<string, mixed> $options
+     *
+     * @return void
+     */
+    protected function addStoreContextForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder->add(static::FORM_STORE_CONTEXT, CollectionType::class, [
+            'entry_type' => StoreContextForm::class,
```

Comment:
See https://github.com/spryker/spryker/pull/10950#discussion_r1638565990, please fix in both PRs.
Half of the code samples should be removed because of this.

Reply:
Changed